src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Group.cs

```diff
@@ -84,14 +63,11 @@ public static Group Synchronized(Group inner)
                 throw new ArgumentNullException(nameof(inner));
 
             // force Captures to be computed.
-
-            CaptureCollection capcoll;
-            Capture dummy;
-
-            capcoll = inner.Captures;
-
+            CaptureCollection capcoll = inner.Captures;
             if (inner._capcount > 0)
-                dummy = capcoll[0];
+            {
+                Capture _ = capcoll[0];
+            }
```

Comment:
Wouldn't it be more intuitive to have an internal ForceInitialized method or something like that on CaptureCollection?